### PR TITLE
ipvlan support and recent optimizes

### PR DESCRIPTION
**Pull request by BSWANG**
* rp_filter auto setting
* fix policy route compatibility on ubuntu
* autoconfig ecs security group 
* ENI secondary ip mode `ipvlan mode` support

### CODE REVIEW

File: daemon/daemon.go [hunk 1]
```diff
@@ -172,6 +173,11 @@ func (networkService *networkService) AllocIP(grpcContext context.Context, r *rp
 		return nil, errors.Wrapf(err, "error get pod resources from db for pod %+v", podinfo)
 	}
 
+	if !networkService.verifyPodNetworkType(podinfo.PodNetworkType) {
+		networkContext.Log().Warnf("unexpect pod network type allocate, maybe daemon mode changed: %+v", podinfo.PodNetworkType)
```
Comment (jzwlqx) [inline daemon/daemon.go]: log and throw is not a good practice

File: daemon/k8s.go [hunk 2]
```diff
@@ -300,20 +316,35 @@ func deserialize(data []byte) (interface{}, error) {
...
 	}
+	return len(podList.Items) > 0, nil
+}
 
+func (k *k8s) GetPod(namespace, name string) (*podInfo, error) {
 	pod, err := k.client.CoreV1().Pods(namespace).Get(name, metav1.GetOptions{})
 	if err != nil {
+		exist, err1 := k.podExist(namespace, name)
+		if err1 != nil {
+			log.Warnf("exist pod failed: %v", err1)
```
Comment (jzwlqx) [inline daemon/k8s.go]: return now

File: plugin/driver/ipvlan.go [hunk 3]
```diff
@@ -1,14 +1,268 @@
...
+	defer func() {
+		if err != nil {
+			deferErr := netlink.LinkDel(&slaveIPVlan)
+			if deferErr != nil {
+				err = errors.Wrapf(err, "ns3Add netlink.LinkDel ipv err %+v", deferErr)
+			}
+		}
+	}()
+
+	slaveLink, err := netlink.LinkByName(hostVeth)
```
Comment (jzwlqx) [inline plugin/driver/ipvlan.go]: change variable name of hostVeth, since it's not a veth

**Comment by BSWANG** [comment]
@jzwlqx All fixed

**Pull request merged**